src/lib/fhirTables.js

```diff
@@ -397,3 +413,196 @@ function obsCompare(a, b) {
   return(effectiveB - effectiveA);
 }
 
+// +--------------------+
+// |    Plan of Care    |
+// +--------------------+
+
+function carePlanHeader() {
+  return (
+    <tr>
+      <th>Status</th>
+      <th>Intent</th>
+      <th>Activities</th>
+      <th>Category</th>
+      <th>Period Start</th>
+      {/* Add headers for other relevant CarePlan properties */}
+    </tr>
+  );
+}
+
+function carePlanRow(r, rmap, dcr) {
+  const status = r.status;
+  const intent = r.intent;
+
+  const activities = futil.joinJSXElements(
+    (r.activity || []).map(activity =>
+      activity.detail && activity.detail.code
+      ? futil.renderCodeableJSX(activity.detail.code, dcr)
+      : null
+    ).filter(activity => activity !== null),
+    ', '
+  );
+
+  const category = futil.joinJSXElements(
+    (r.category || []).map(c =>
+      c.text ? futil.renderCodeableJSX(c, dcr) : null
+    ).filter(c => c !== null),
+    ', '
+  );
+
+  const period = r.period ? futil.renderPeriod(r.period) : "";
+
+  return (
+    <tr key={r.id}>
+      <td>{status}</td>
+      <td>{intent}</td>
+      <td>{activities}</td>
+      <td>{category}</td>
+      <td>{period}</td>
+      {/* Render other relevant CarePlan properties as table cells */}
+    </tr>
+  );
+}
+
+// +--------------------+
+// |      Consent       |
+// +--------------------+
+
+function consentHeader() {
+  return (
+    <tr>
+      <th>Status</th>
+      <th>Scope</th>
+      <th>Category</th>
+      <th>Date/Time</th>
+      <th>Policy Rule</th>
+      <th>Provision Period</th>
+      <th>Organization</th>
+    </tr>
+  );
+}
+
+function consentRow(r, rmap, dcr) {
+  const status = r.status || "N/A";
+  const scopeDisplay = r.scope && Array.isArray(r.scope.coding) && r.scope.coding.length > 0
+    ? futil.renderCodeableJSX(r.scope, dcr)
+    : "";
+  const categoryDisplay = r.category && Array.isArray(r.category) && r.category.length > 0
+    ? futil.renderCodeableJSX(r.category[0], dcr)
+    : "";
+  const dateTime = r.dateTime
+    ? futil.renderDateTime(r.dateTime)
+    : "";
+  const policyRule = r.policyRule && Array.isArray(r.policyRule.coding) && r.policyRule.coding.length > 0
+    ? futil.renderCodeableJSX(r.policyRule, dcr)
+    : "";
+  const provisionPeriod = r.provision && r.provision.period
+    ? futil.renderPeriod(r.provision.period)
+    : "";
+  const organization = r.organization && Array.isArray(r.organization) && r.organization.length > 0
+    ? futil.renderOrganization(r.organization[0], dcr)
+    : "";
+
+  return (
```

Comment:
there are a few of these codeable types that have code systems we should add to the top of codes.js:
* http://terminology.hl7.org/CodeSystem/consentpolicycodes
* http://terminology.hl7.org/CodeSystem/consentscope

There is another one, but it's a bit tougher:
* https://hl7.org/fhir/R4/valueset-consent-category.html ... it's a "ValueSet" that contains codes from a bunch of different places. We really need to implement this in codes.js but it's not done yet. I'll make a separate issue for this, but if you could add the two simple codesets above (let me know if not clear how to do from the existing entries, no problem if so!) that would be great.

Reply:
I took a shot at adding the codes but not 100% if I missed any other steps here... :grin: